types/react-instantsearch-core/index.d.ts

```diff
@@ -0,0 +1,341 @@
+// Type definitions for react-instantsearch-core 5.2
+// Project: https://community.algolia.com/react-instantsearch/
+// Definitions by: Gordon Burgett <https://github.com/gburgett>
+//                 Justin Powell <https://github.com/jpowell>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.9
+
+import * as React from 'react';
+
+import { ConnectedComponentType, Hit, SearchResults, SearchState, BasicDoc, AlgoliaError } from './types';
+
+export { Hit, HighlightResult, SearchResults, SearchState, BasicDoc, AlgoliaError } from './types';
+
+// Core
+/**
+ * Creates a specialized root InstantSearch component. It accepts
+ * an algolia client and a specification of the root Element.
+ * @param defaultAlgoliaClient - a function that builds an Algolia client
+ * @param root - the defininition of the root of an InstantSearch sub tree.
+ * @returns an InstantSearch root
+ */
+export function createInstantSearch(defaultAlgoliaClient: (...args: any[]) => any, root: any): any;
```

Comment:
The type of client could be () => void since that's its expected to be a function that can take any number of arguments and return anything

Reply:
type of root needs to be object and not any.

Reply:
https://community.algolia.com/react-instantsearch/server-side-renderings/createInstantSearch.html seems to have more information.

Reply:
A client object (not sure if relevant here) looks like this:

```js
interface SearchClient {
  search(requests: Request[]): Promise<Result[]>,
  search(requests: Request[]): Promise<Result[]>,
}
```

The Request can be derived from `algoliasearch.QueryParameters` (also on DT) and `algoliasearch.SearchForFacetValues.Parameters`

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/algoliasearch/index.d.ts#L1423 and https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/algoliasearch/index.d.ts#L1101-L1420